traceevents/src/main/java/com/tomtom/kotlin/traceevents/TraceEvent.kt

```diff
@@ -42,8 +45,21 @@ data class TraceEvent(
     val interfaceName: String,
     val stackTraceHolder: Throwable?,
     val eventName: String,
-    val args: Array<Any?>
+    val args: Array<Any?>,
+    val parameterNames: Array<String>?
 ) {
+    init {
+        check(parameterNames == null || args.size == parameterNames.size)
+    }
+
+    /**
+     * Get the method parameters as a map that maps the parameter name to its value.
+     * Returns an empty map for parameterless methods and `null` if 1 or more parameter names are not
+     * available for some reason.
+     */
+    fun getNamedParametersMap() = if (parameterNames == null) null else
+        parameterNames.indices.associateBy({ parameterNames[it] }, { args[it] })
```

Comment:
`if (x == null) null else` is typically a sign that a `?` can reduce clutter:
```suggestion
    fun getNamedParametersMap() = 
        parameterNames?.indices?.associateBy({ parameterNames[it] }, { args[it] })
```